instance/models/mixins/openedx_monitoring.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2016 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Open edX instance monitoring mixin
+"""
+
+# Imports #####################################################################
+
+from django.conf import settings
+from django.db import models
+
+from instance import newrelic
+
+
+# Classes #####################################################################
+
+class OpenEdXMonitoringMixin:
+    """
+    Mixin that sets up availability monitoring for Open edX instances.
+    """
+    def enable_monitoring(self):
+        """
+        Enable monitoring on this instance.
+        """
+        if not settings.NEWRELIC_ADMIN_USER_API_KEY:
+            return
+
+        # Delete existing monitors if they don't monitor this instance's
+        # public urls
+        monitors = newrelic.get_synthetics_monitors()
```

Comment:
@omarkhan Nit: Maybe we should add a `uri` field to `NewRelicAvailabilityMonitor` model so that we don't have to fetch the full global list of monitors every time?